code-challenges.test.js

```diff
@@ -0,0 +1,159 @@
+        expect(dividedBy3(object2.number)).toEqual("0 is divisible by three")
+        expect(dividedBy3(object3.number)).toEqual("-7 is not divisible by three")
+    })
+})
+
+const dividedBy3 = (object) => {
+    if (object % 3 === 0) {
+    return `${object} is divisible by three`
+    } else {
+    return `${object} is not divisible by three`
+    }
+}
+
+
+
+
+// 
+
+// --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+
+// b) Create the function that makes the test pass.
+
+// Pseudo code:
+
+//  Write a function that take in the array and add the map (HOF) to iterate in the array
+
+//  In the map of the HOF write a function that will take the value that's iterating and split the strings into individual strings of letters
+
+//  create another map HOF that will iterate each letter and add a conditional statement to return the 0 index in the string uppercased and the rest of the string 
+
+//  Add the .join('') with single quotations to join the string "letters" back to string "words"
+
+const arrayWords = (array) => {
+    return array.map(value => value.split('').map((value, index) => { 
+        if (index == 0)
+        return value.toUpperCase()
+        else {
+            return value
+        }
+    }).join(''))     
+}
+
+console.log(arrayWords(randomNouns1))
+
+describe("arrayWords", () => {
+    it("Capitalize the first letter in the string", () => {
+        expect(arrayWords(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(arrayWords(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+})
+
+
+
+// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const vowelTester1 = "learn"
+// Expected output: 1
+const vowelTester2 = "academy"
+// Expected output: 0
+const vowelTester3 = "challenges"
+// Expected output: 2
+
+describe("vowelChecker", () => {
+    it("returns the first instance of a vowel in the word", () => {
+        expect(vowelChecker(vowelTester1)).toEqual(1)
+        expect(vowelChecker(vowelTester2)).toEqual(0)
+        expect(vowelChecker(vowelTester3)).toEqual(2)
+    })
+})
+
+// b) Create the function that makes the test pass.
+
+// Pseudo code:
+
+// create function to check first instance of a vowel 
+// make an array of vowels
+// for loop through string and check for vowels
+// Set conditions to check all characters within word for vowels
```

Comment:
But how? Be sure to describe how your code is doing this rather than just saying what you want to do